tests/test-wibox-shape.lua

```diff
@@ -18,14 +18,107 @@ local wb = wibox {
 wb:geometry(screen[1].geometry)
 wb.visible = true
 
-runner.run_steps({
+local count = 0
+
+local steps = {
     function()
         assert(wb.shape == shape.powerline)
         assert(wb.shape_bounding) -- This is a memory leak! Don't copy!
         if was_drawn then
             return true
         end
+    end,
+    -- Remove the shape and test the input
+    function()
+        wb.shape = nil
+        wb.input_passthrough = false
+        wb:connect_signal("button::press", function()
+            count = count + 1
+        end)
+
+        wb:geometry {
+            x      = 0,
+            y      = 100,
+            width  = 101,
+            height = 101,
+        }
+        wb.border_width = 0
+
+        return true
     end
-})
+}
+
+-- Emulate a click.
+-- Each pair of the `...` corresponds to a point.
+local function click(...)
+    local args = {...}
+    table.insert(steps, function()
+        for i = 0, math.floor(#args/2)-1 do
+            local x, y = args[i*2+1], args[i*2+2]
+            mouse.coords{x=x, y=y}
+            assert(mouse.coords().x == x and mouse.coords().y == y)
+            root.fake_input("button_release", 1)
+            root.fake_input("button_press", 1)
+        end
+
+        return true
+    end)
+end
+
+local function check_count(cnt)
+    table.insert(steps, function()
+        awesome.sync()
```

Comment:
I think this `sync` should be before the `return true` in `click` and then this function can just `return cnt == count`, but feel free to ignore me here....